### PR TITLE
use more === in tfuncs

### PR DESCRIPTION
**Pull request by oxinabox**
I don't know that this will have much effect on performance.
In general compiler reasons better about `===` but since this whole this in in `@nospecialize` idk about that.
Also it avoids potential invalidations, but since it is `Core.==` it is probably not invalidate.
But still it is nice for consistency.
We mostly use `===` throughout this file, this changed to almost entirely use that.

It also tweaks a few comments,
and makes a bit of spacing more consistent with the rest of the file.

(I was just reading though the file looking to understand it better)

### CODE REVIEW

File: base/compiler/tfuncs.jl [hunk 1]
```diff
@@ -717,7 +718,7 @@ function getfield_nothrow(@nospecialize(s00), @nospecialize(name), @nospecialize
         s.abstract && return false
         # If all fields are always initialized, and bounds check is disabled, we can assume
         # we don't throw
-        if bounds_check_disabled && !isvatuple(s) && s.name !== NamedTuple.body.body.name && fieldcount(s) == s.ninitialized
+        if bounds_check_disabled && !isvatuple(s) && s.name !== NamedTuple.body.body.name && fieldcount(s) === s.ninitialized
```
Comment (vtjnash) [inline base/compiler/tfuncs.jl:721]: This makes it more brittle:
```
julia> typeof(Int).ninitialized |> typeof
Int32

julia> fieldcount(Int) |> typeof
Int64
```

File: base/compiler/tfuncs.jl [hunk 2]
```diff
@@ -320,8 +320,8 @@ function sizeof_nothrow(@nospecialize(x))
             return isinline
         end
         isa(x, DataType) || return false
-        x.layout == C_NULL && return false
-        (datatype_nfields(x) == 0 && !datatype_pointerfree(x)) && return false
+        x.layout === C_NULL && return false
+        (datatype_nfields(x) === 0 && !datatype_pointerfree(x)) && return false
```
Comment (vtjnash) [inline base/compiler/tfuncs.jl:324]: Same here:
```
julia> Base.datatype_nfields(Int) |> typeof
UInt32
```

File: base/compiler/tfuncs.jl (hunk 1 above)
Comment (oxinabox) [reply base/compiler/tfuncs.jl:721]: Huh, TIL.
i thought both of these were certain to be `Int`

**Comment by Keno** [comment]
I don't think there is much point to changing all the integer comparisons to `===`. The fact that it's an integer on either side should be well inferred.

**Comment by vtjnash** [comment]
Hopefully nobody is redefining `Core.Compiler.:(+)(::Int, ::Int)`, so while this isn't wrong, it's just not really an improvement either.

**Pull request closed without merging**